content/blog/2025-07-07-user-defined-parquet-indexes.md

```diff
@@ -0,0 +1,542 @@
+---
+layout: post
+title: Extending Apache Parquet with User Defined Indexes to Accelerate Query Processing with DataFusion
+date: 2025-07-07
+author: Qi Zhu, Jigao Luo, and Andrew Lamb
+categories: [features]
+---
+<!--
+{% comment %}
+Licensed to the Apache Software Foundation (ASF) under one or more
+contributor license agreements.  See the NOTICE file distributed with
+this work for additional information regarding copyright ownership.
+The ASF licenses this file to you under the Apache License, Version 2.0
+(the "License"); you may not use this file except in compliance with
+the License.  You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+{% endcomment %}
+-->
+
+It’s a common misconception that [Apache Parquet] files can only store basic Min/Max/Null Count statistics and Bloom filters, and that adding anything "smarter" requires a change to the specification or an entirely new file format. In fact, footer metadata and offset based addressing already provide everything needed to embed user defined index structures within Parquet Files without breaking compatibility with other Parquet readers. 
```

Comment:
FYI I am very pleased with this intro as I think it succinctly summarizes the value of this blog and will get people excited

<img width="828" alt="Screenshot 2025-07-06 at 9 35 24 AM" src="https://github.com/user-attachments/assets/a86a5d3f-c90e-426a-b75d-493b78adb1da" />